go-controller/pkg/util/multi_network.go

```diff
@@ -577,3 +577,24 @@ func GetPodNADToNetworkMapping(pod *kapi.Pod, nInfo NetInfo) (bool, map[string]*
 	}
 	return true, networkSelections, nil
 }
+
+type networkControllerTopologyNotSupportedError struct {
+	msg string
+}
+
+func (nctnse networkControllerTopologyNotSupportedError) Error() string {
+	return nctnse.msg
+}
+
+// NewNetworkControllerTopologyNotSupportedError returns an error for
+// network controller not supported for the network topology
+func NewNetworkControllerTopologyNotSupportedError(format string, args ...interface{}) error {
+	return networkControllerTopologyNotSupportedError{msg: fmt.Sprintf(format, args...)}
+}
+
+// IsNetworkControllerTopologyNotSupportedError returns true if the error indicates the network
+// controller is not supported for the topology
+func IsNetworkControllerTopologyNotSupportedError(err error) bool {
+	_, ok := err.(networkControllerTopologyNotSupportedError)
+	return ok
+}
```

Comment:
nit: 

I would define this error in `network_attach_def_controller.go` as it is very specific (which is fine, but as opposed to a `TopologyNotSupportedError` which would be more generic).

Also not that it is not very useful to build an specific error type with an open ended format string.

You would either have an error that would take specific parameters to force specific information being provided:
`func NewNetworkControllerTopologyNotSupportedError(topology string)`
and then decide the appropriate string internally
```
func (nctnse networkControllerTopologyNotSupportedError) Error() string {
	return fmt.Sprintf("No cluster network controller to manage topology %s", nctnse.topology)
}
```

or either have pretty much an simpler easier to handle error
`NetworkControllerTopologyNotSupportedError = errors.New("No cluster network controller to manage topology")`
and relay the responsibility to the user in providing the appropriate context information

Reply:
Done